public/components/utils/EditableDropdown.js

```diff
@@ -0,0 +1,69 @@
+import React, { PropTypes } from 'react';
+
+class EditableDropdown extends React.Component {
+
+  static propTypes = {
+    name: PropTypes.string.isRequired,
+    values: PropTypes.object.isRequired,
+    selectedValue: PropTypes.string.isRequired,
+    onChange: PropTypes.func.isRequired
+  };
+
+  state = {
+    editable: false
+  }
+
+  enableEditing = () => {
+    this.setState({
+      editable: true
+    });
+  }
+
+  disableEditing = () => {
+    this.setState({
+      editable: false
+    });
+  }
+
+  prepareOptions = () => {
+    const values = this.props.values;
+    let options = [];
+
+    for (let prop in values) {
```

Comment:
Can we make this more functional, I think a filter and map or a reduce would work well.